app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -534,9 +535,10 @@ private void VibrateNotifyMakeNoise(Context context, AlertType alert, String bgV
                 .setDeleteIntent(snoozeIntent(context, minsFromStartPlaying));
 
         if (profile != ALERT_PROFILE_VIBRATE_ONLY && profile != ALERT_PROFILE_SILENT) {
-            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES) {
-                // Before this, we only vibrate...
-                float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
+            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES || (minsFromStartPlaying > Vibrating_Calc && !Pref.getBoolean("delay_ascending_3min", true))) {
```

Comment:
This could be rewritten to
`if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES || (Vibrating_Calc > 0 && minsFromStartPlaying > Vibrating_Calc)) {`

Reply:
But, it will make the code harder to understand.  
Someone who is looking at this for the first time, will have to go look for Vibrating_Calc to see what its value is in order to find out that its value depends on a setting.  Whereas what we have now right away shows that we are checking a setting.
I see that if I make the change you have suggested, the line will become a little bit shorter.  But, do you think it is worth the price of making the code harder to understand?

Reply:
I don't think it make it harder to understand. IMHO, the opposite is the case and code duplication is avoided. One has to lookup `Vibrating_Calc` always to understand the code. From this time, it knows that it depends on a preference.

Avoiding code duplication does not only save space and improves readability. It also increases maintenance and reduces the likelihood of errors. 

@jwoglom What do you think about code readability in this context?

Reply:
Some day, someone may need to rewrite the formula on line 539.  And they may need to change the values of Vibrating_Calc.  Then, they will have one more restriction they will have to worry about if we make this suggested edit, which will be the logic in the suggested change.

But, the intended logic is extremely simple; either the user has enabled the setting or not.  It is true that right now, Vibrating_Calc only has two possible values and one of them is greater and the other is smaller than 0.  But, that function of Vaibrating_Calc does not constitute that condition will always be true.

The suggested change adds a restriction to the code, on the values of Vibrating_Calc.  
I chose to write the formula (line 539) the way I did.  There is no reason it cannot be written in a simpler way.  But, imposing more restrictions on it seems to be unnecessary.

Reply:
I think there is an easy solution without introducing a new variable, just with an if-else clause like

```java
float volumeFrac = 0;
if (!Pref.getBoolean("delay_ascending_3min", true)) {
    volumeFrac = (float) (minsFromStartPlaying + 1)/6;
} else {
    if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES) {
                // Before this, we only vibrate...
                volumeFrac = (float)(minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
   }
}
if (volumeFrac > 0) {
    ...
}
```

Reply:
I like that.  I need time to test it.
Thanks a lot.

Reply:
I changed a lot.  There is no new formula any longer.
My phone would make no noise when the volume was lower than 30%.
So, I changed the behavior and updated the description.